src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -4411,83 +4423,93 @@ trait Types
    *  Return `x` if the computation succeeds with result `x`.
    *  Return `NoType` if the computation fails.
    */
-  def mergePrefixAndArgs(tps: List[Type], variance: Variance, depth: Depth): Type = tps match {
-    case tp :: Nil => tp
-    case TypeRef(_, sym, _) :: rest =>
-      val pres = tps map (_.prefix) // prefix normalizes automatically
+  def mergePrefixAndArgs(tps0: List[Type], variance: Variance, depth: Depth): Type = {
+    var tparams = mutable.ListBuffer[Symbol]()
+    val tps = tps0.flatMap {
+      case rt: RefinedType if isIntersectionTypeForLazyBaseType(rt) => rt.parents
+      case ExistentialType(qs, underlying) =>
+        tparams ++= qs
+        underlying match {
+          case rt: RefinedType if isIntersectionTypeForLazyBaseType(rt) => rt.parents
+          case tp => tp :: Nil
+        }
+      case tp => tp :: Nil
+    }
```

Comment:
Unwrap all the existentials, IntersectionTypeForLazyBaseType-s, and existentials-over-IntersectionTypeForLazyBaseType.